frankenphp.go

```diff
@@ -468,19 +471,16 @@ func go_handle_request() bool {
 		return false
 
 	case r := <-requestChan:
-		h := cgo.NewHandle(r)
-		r.Context().Value(handleKey).(*handleList).AddHandle(h)
-
 		fc, ok := FromContext(r.Context())
 		if !ok {
 			panic(InvalidRequestError)
 		}
 		defer func() {
 			maybeCloseContext(fc)
-			r.Context().Value(handleKey).(*handleList).FreeAll()
 		}()
 
-		if err := updateServerContext(r, true, 0); err != nil {
+		rh, err := updateServerContext(r, true, 0)
+		if err != nil {
 			panic(err)
```

Comment:
nit, but a leak may happen if `recover` is used by library users:

```suggestion
			rh.Delete()
			panic(err)
```

Alternatively, we could put `defer rh.Delete()` just before the if block, but it may (or may not, it depends on how the compiler manages to optimize this call to `defer`) cost one extra allocation.